sensing/src/main/java/com/google/android/sensing/impl/SensingEngineImpl.kt

```diff
@@ -179,18 +187,21 @@ internal class SensingEngineImpl(
 
     // Step 1: Delete db records
     database.deleteRecordsInCapture(captureId)
-    // Step 2: delete the captureFolder
-    val captureFile = File(context.filesDir, captureInfo.captureFolder)
-    val parentFile = captureFile.parentFile
-    val deleted: Boolean
-    withContext(Dispatchers.IO) {
-      deleted = captureFile.deleteRecursively()
-      // delete Participant's folder if there are no data
-      if (parentFile?.list()?.isEmpty() == true) {
-        parentFile.delete()
+    if (isRecapture) {
+      // Step 2: delete the captureFolder
+      val captureFile = File(context.filesDir, captureInfo.captureFolder)
+      val parentFile = captureFile.parentFile
+      val deleted: Boolean
+      withContext(Dispatchers.IO) {
+        deleted = captureFile.deleteRecursively()
+        // delete Participant's folder if there are no data
+        if (parentFile?.list()?.isEmpty() == true) {
+          parentFile.delete()
+        }
       }
+      return deleted
     }
-    return deleted
+    return false
```

Comment:
Question: What does this return flag signify?

Reply:
Actually no return flag is required as we are not processing anything further based on delete success flag so can eliminate it by changing return type of that method,

Reply:
Ok thanks for the clarification, Please go ahead